src/utils/FeatureDetectionUtils.ts

```diff
@@ -1,3 +1,13 @@
+// Necessary because of https://github.com/Microsoft/TypeScript/issues/14062
+// Introduced in typescript 2.2, should be fixed in 2.3
+interface IDocument {
+  implementation: {
+    hasFeature: any;
+  };
+}
+
+declare var document: IDocument;
+
 export class FeatureDetectionUtils {
```

Comment:
Yeah, but it's not documented.

Anything that is not publicly documented, it's fair game to remove them.